src/Mesh/Interpolation.jl

```diff
@@ -1439,36 +1453,96 @@ function interpolate_cubed_sphere_CUDA!(
     return nothing
 end
 #--------------------------------------------------------
+"""
+    project_cubed_sphere!(intrp_cs::InterpolationCubedSphere{FT}, 
+                                 v::AbstractArray{FT}, 
+                              uvwi::Tuple{Int,Int,Int}) where {FT <: AbstractFloat}
+
+This function projects the velocity field along unit vectors in radial, lat and long directions for cubed spherical shell geometry.
+
+# Fields
+ - `intrp_cs` Initialized cubed sphere structure
+ - `v` Array consisting of velocity field on the discontinuous Galerkin grid
+ - `uvwi`  Tuple providing the column numbers for u, v and w in the array. These columns will be replaced with projected velocity fields.
+"""
+function project_cubed_sphere!(
+    intrp_cs::InterpolationCubedSphere{FT},
+    v::AbstractArray{FT},
+    uvwi::Tuple{Int, Int, Int},
+) where {FT <: AbstractFloat}
+    # projecting velocity onto unit vectors in rad, lat and long directions
+    # assumes u, v and w are located in columns specified in vector uvwi
+    @assert length(uvwi) == 3 "length(uvwi) is not 3"
+    lati = intrp_cs.lati
+    longi = intrp_cs.longi
+    lat_grd = intrp_cs.lat_grd
+    long_grd = intrp_cs.long_grd
+    _ρu = uvwi[1]
+    _ρv = uvwi[2]
+    _ρw = uvwi[3]
+    np_tot = size(v, 1)
+
+    device = typeof(v) <: Array ? CPU() : CUDA()
+    if device == CPU()
```

Comment:
```suggestion
    if v isa Array
```
then you can get rid of `CPU` and `CUDA`

Reply:
Using ```import KernelAbstractions: CPU, CUDA``` per our conversation.